### DISCUSSION TITLE
Add rabbitmq-delayed-message-exchange plugin to docker, rabbitmq-plugins list already has the E note, but can not find the x-delayed-message type switch

### DISCUSSION THREAD

**Discussion by Jinyilong** [rabbitmq/rabbitmq-delayed-message-exchange#249] [General]
### Describe the bug

 can not find the x-delayed-message type switch

### Reproduction steps

1.docker cp /root/rabbitmq_delayed_message_exchange-3.8.17.8f537ac.ez  mq:/plugins/
2.docker exec  -it 46b18d0b7a82 /bin/bash
3.cd /plugins
4.rabbitmq-plugins enable rabbitmq_delayed_message_exchange
5.rabbitmq-plugins list
6.Listing plugins with pattern ".*" ...
 Configured: E = explicitly enabled; e = implicitly enabled
 | Status: * = running on rabbit@mq1
 |/
[  ] rabbitmq_amqp1_0                  3.8.26
[  ] rabbitmq_auth_backend_cache       3.8.26
[  ] rabbitmq_auth_backend_http        3.8.26
[  ] rabbitmq_auth_backend_ldap        3.8.26
[  ] rabbitmq_auth_backend_oauth2      3.8.26
[  ] rabbitmq_auth_mechanism_ssl       3.8.26
[  ] rabbitmq_consistent_hash_exchange 3.8.26
[E ] rabbitmq_delayed_message_exchange 3.8.9+1.g8f537ac
[  ] rabbitmq_event_exchange           3.8.26
[  ] rabbitmq_federation               3.8.26
[  ] rabbitmq_federation_management    3.8.26
[  ] rabbitmq_jms_topic_exchange       3.8.26
[E*] rabbitmq_management               3.8.26
[e*] rabbitmq_management_agent         3.8.26
[  ] rabbitmq_mqtt                     3.8.26
[  ] rabbitmq_peer_discovery_aws       3.8.26
[  ] rabbitmq_peer_discovery_common    3.8.26
[  ] rabbitmq_peer_discovery_consul    3.8.26
[  ] rabbitmq_peer_discovery_etcd      3.8.26
[  ] rabbitmq_peer_discovery_k8s       3.8.26
[E*] rabbitmq_prometheus               3.8.26
[  ] rabbitmq_random_exchange          3.8.26
[  ] rabbitmq_recent_history_exchange  3.8.26
[  ] rabbitmq_sharding                 3.8.26
[  ] rabbitmq_shovel                   3.8.26
[  ] rabbitmq_shovel_management        3.8.26
[  ] rabbitmq_stomp                    3.8.26
[  ] rabbitmq_top                      3.8.26
[  ] rabbitmq_tracing                  3.8.26
[  ] rabbitmq_trust_store              3.8.26
[e*] rabbitmq_web_dispatch             3.8.26
[  ] rabbitmq_web_mqtt                 3.8.26
[  ] rabbitmq_web_mqtt_examples        3.8.26
[  ] rabbitmq_web_stomp                3.8.26
[  ] rabbitmq_web_stomp_examples       3.8.26
7.docker restart 46b18d0b7a82
8. But I still don't have this type in my switch
9. connection error; protocol method: #method<connection.close>(reply-code=503, reply-text=COMMAND_INVALID - unknown exchange type 'x-delayed-message', class-id=40, method-id=10)
...

### Expected behavior

I want my mq to recognize a plugin

**Comment by Jinyilong** [comment]
package com.it.jyl.rabbitmq.ttl.TtlQueueConfig;

import org.springframework.amqp.core.Binding;
import org.springframework.amqp.core.BindingBuilder;
import org.springframework.amqp.core.CustomExchange;
import org.springframework.amqp.core.Queue;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.HashMap;
import java.util.Map;

@Configuration
public class DelayedQueueConfig {
    //延迟队列
    public static final String DELAYED_QUEUE_NAME = "delayed.queue";
    //延迟交换机
    public static final String DELAYED_EXCHANGE_NAME = "delayed.exchange";
    //路由key
    public static final String DELAYED_ROUTING_KEY = "delayed.routingkey";

    //申明队列
    @Bean("delayedQueue")
    public Queue delayedQueue() {
        return new Queue(DELAYED_QUEUE_NAME);
    }

    //自定义交换机 我们在这里定义的是一个延迟交换机
    @Bean("delayedExchange")
    public CustomExchange delayedExchange() {
        Map<String, Object> args = new HashMap<>();
        //自定义交换机的类型 ????
        args.put("x-delayed-type", "direct");
        return new CustomExchange(DELAYED_EXCHANGE_NAME, "x-delayed-message", true, false,
                args);
    }

    @Bean
    public Binding bindingDelayedQueue(@Qualifier("delayedQueue") Queue queue,
                                       @Qualifier("delayedExchange") CustomExchange
                                               delayedExchange) {
        return

                BindingBuilder.bind(queue).to(delayedExchange).with(DELAYED_ROUTING_KEY).noargs();
    }
}

this is my DelayedQueueConfig

**Comment by x1ld3n** [comment]
Hi. I have the exact same issue with RabbitMQ 3.12.x and plugin version 3.12.0

**Comment by Jinyilong** [comment]
> 
I started with version 3.12 and it didn't work ，then use 3.8  it didn't work too

**Comment by x1ld3n** [comment]
For me it's working with RabbitMQ 3.11.18 / 3.11.19 and plugin version 3.11.1, anything above it's not working.

**Comment by Jinyilong** [comment]
root@mq1:/plugins# rabbitmq-plugins enable rabbitmq_delayed_message_exchange
Enabling plugins on node rabbit@mq1:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_prometheus
  rabbitmq_web_dispatch
Applying plugin configuration to rabbit@mq1...
The following plugins have been enabled:
  rabbitmq_delayed_message_exchange

nothing to do.
why IT is nothing to do

**Comment by Jinyilong** [comment]
> For me it's working with RabbitMQ 3.11.18 / 3.11.19 and plugin version 3.11.1, anything above it's not working.

root@mq1:/plugins# rabbitmq-plugins enable rabbitmq_delayed_message_exchange
Enabling plugins on node rabbit@mq1:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_prometheus
  rabbitmq_web_dispatch
Applying plugin configuration to rabbit@mq1...
The following plugins have been enabled:
  rabbitmq_delayed_message_exchange

started 1 plugins.
This is a version problem , I changed so many versions that I finally  RabbitMQ 3.8.26  rabbitmq_delayed_message_exchange-3.8.9-0199d11c.ez

**Comment by michaelklishin** [comment]
RabbitMQ 3.8 has reached EOL year ago.

Take a look at the existing discussions and you will quickly find some very similar ones where the issue came down to the plugin file permissions, or a broker/plugin version mismatch.
